### ISSUE TITLE
Allow arbitrary "dedup unit" sizes in RGBGFX

### ISSUE THREAD

**Issue by ISSOtm** [gbdev/rgbds#1005]
This would be useful for "8×16 OAM mode", mainly, but also metatile setups that require four unique tile IDs (usually those that don't go through an extra indirection layer).

**Comment by CharlesVanEeckhout** [comment]
Image from #1412 that shows what order to read the tiles
![image](https://github.com/gbdev/rgbds/assets/99271432/065da233-f249-4888-90a2-52c1a76cc7a7)

**Comment by ISSOtm** [comment]
The idea I had in mind was that `-U` would affect the way data is generated, but not how it's emitted beyond ordering.

**Comment by Rangi42** [comment]
Regarding 8x16px OAM tile support: OAM sprites are often composed of multiple objects, arranged in arbitrary positions, not just on a grid. We can't support overlapping objects, but maybe with a `-B`ackground color, we could support non-grid tiles? For instance, these are 8x16px tiles, aligned on an 8x8px grid:

<img width="320" height="64" alt="Image" src="https://github.com/user-attachments/assets/10f724a7-13a9-4930-8476-5752f2580012" />

<img width="320" height="32" alt="Image" src="https://github.com/user-attachments/assets/f189461f-ab4a-41b2-afb7-5fac74749d77" />

Ideally I'd want to convert them with `rgbgfx --background #abcdef --unique-tiles --oam` to recreate [hooh_gold.png](https://github.com/pret/pokegold/blob/master/gfx/title/hooh_gold.png) and [lugia_silver.png](https://github.com/pret/pokegold/blob/master/gfx/title/lugia_silver.png).

**Comment by ISSOtm** [comment]
No, because a lot of the time, OBJs are not laid out on a grid:

Naive | Packed
-|-
<img width="512" height="448" alt="example of using OBJs like a grid" src="https://github.com/user-attachments/assets/3e93feea-e236-4d71-baa9-ea0dc3698f82" /> | <img width="512" height="448" alt="example of using OBJs more effectively" src="https://github.com/user-attachments/assets/6028e5be-6cec-43bb-b51d-f57333be023d" />

I think we should go with a solution like what Pino developed for *Libbet*: some kind of description of where cels are located, and how they're sliced. (Ideally, we'd find a way to support “layered” gfx as well, where OBJs with different palettes are used to provide more colours than normal; but we may rule that out as too difficult to implement.)

**Comment by Rangi42** [comment]
Would it be sufficient to allow padding `-L/--slice` multiple times? Easy enough to pass `-o foo.2bpp @foo.slices foo.png` if you want a separate file. The order of `-L` arguments would of course be the order in which they're processed.

That could sort of *sometimes* handle overlapping objects, since slices can overlap:

<img width="12" height="12" alt="Image" src="https://github.com/user-attachments/assets/be95238b-c581-4bee-864e-c2c8e3b4887b" />
(`-L 0,0:1,1 -L 4,4:1,1`)

But if the >4 colors are literally all in the same tile, it wouldn't work, even though that's something OAM objects can overlap to do:

<img width="8" height="8" alt="Image" src="https://github.com/user-attachments/assets/cff0cfd9-99e4-4737-996a-aad8811516a3" />

Still, I think multiple slices would cover enough use cases to be worth doing. I'll open a separate issue for 1.x if that sounds good.

**Comment by ISSOtm** [comment]
I'm not entirely sure about multiple slices, but the best way to resolve that is certainly trying things out and seeing. (Just one remark: considering that for all arguments thus far, passing them multiple times causes silent overriding, I'd rather the feature use a single `--slice` rather than multiple. Though that will require introducing an extra separator, so perhaps a `--slices-from-file` would be more practical and realistic?)

**Comment by Rangi42** [comment]
I wouldn't want to *require* a separate file for multiple slices. Another separator sounds fine to me:
```
--slice 0,0:1,1;4,4:1,1
```
With whitespace allowed for clarity:
```
--slice "0,0:1,1; 4,4:1,1"
```

**Comment by Rangi42** [comment]
Until/unless RGBGFX supports 8x16px tiles for OAM sprites, you can also use [SuperFamiconv](https://github.com/Optiroc/SuperFamiconv/) (`-W 8 -H 16`).
